raft-tech/TANF-app
# 2411 metadata parsed datafiles

## Summary of Changes
_Provide a brief summary of changes_
Pull request closes #2411 _

## How to Test
To test, use the following commands to start the app:

```
cd tdrs-frontend && docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d
cd tdrs-backend && docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d 
```

1. Open http://localhost:3000/ and sign in.
1. Load a couple of test files to add datafile errors
1. Make sure you have an admin user account and login to the admin page
1. Proceed to data file page from the left panel

You will see the loaded datafiles and the added two columns as shown below:

![Screenshot 2023-09-27 at 10 35 54 AM](https://github.com/raft-tech/TANF-app/assets/97037188/eb0b2a09-886b-43af-a51a-7bc1ddd28391)


## Deliverables
Added new 'program type' filter to datafile list page, as well as two new columns:

- DATA FILE SUMMARY
- ERROR REPORT LINK


### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:

- [x] Parsed datafiles are filterable on: STT, Year, Quarter, Program Type
- [x] Admin view of parsed datafiles contain: file status, case totals, err report link, header enc flag, link to STT object, link to originating DataFile
- [x] Clickable obj link is not an incrementing integer

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
+ Are code coverage minimums met?

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)


### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)



### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)


### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)


### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)


### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
